core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -161,8 +162,15 @@ static void disableShutdownHook() {
      * Acquires an {@link EventLoop} that is expected to handle a connection to the specified {@link Endpoint}.
      * The caller must release the returned {@link EventLoop} back by calling {@link ReleasableHolder#release()}
      * so that {@link ClientFactory} utilizes {@link EventLoop}s efficiently.
+     *
+     * @param sessionProtocol the {@link SessionProtocol} of the connection
+     * @param endpointGroup the {@link EndpointGroup} where {@code endpoint} belongs to.
+     * @param endpoint the {@link Endpoint} where a request is being sent.
+     *                 {@code null} if the {@link Endpoint} is not known yet.
      */
-    ReleasableHolder<EventLoop> acquireEventLoop(Endpoint endpoint, SessionProtocol sessionProtocol);
+    ReleasableHolder<EventLoop> acquireEventLoop(SessionProtocol sessionProtocol,
+                                                 EndpointGroup endpointGroup,
+                                                 @Nullable Endpoint endpoint);
```

Comment:
Maybe, Is this a breaking change? Need a `breaking-changes` label?

Reply:
Good idea. Added the label.